camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/mapred/EtlInputFormatTest.java

```diff
@@ -0,0 +1,100 @@
+package com.linkedin.camus.etl.kafka.mapred;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import kafka.javaapi.TopicMetadata;
+import kafka.javaapi.TopicMetadataRequest;
+import kafka.javaapi.TopicMetadataResponse;
+import kafka.javaapi.consumer.SimpleConsumer;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.easymock.EasyMock;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class EtlInputFormatTest {
+
+  private static final String DUMMY_VALUE = "dummy:1234";
+  
+  @Test
+  public void testWithOneRetry() {
+    List<Object> mocks = new ArrayList<Object>();
+    
+    Configuration configuration = EasyMock.createMock(Configuration.class);
+    mocks.add(configuration);
+    EasyMock.expect(configuration.get(EasyMock.anyString())).andReturn(DUMMY_VALUE).anyTimes();
+    
+    JobContext jobContext = EasyMock.createMock(JobContext.class);
+    mocks.add(jobContext);
+    EasyMock.expect(jobContext.getConfiguration()).andReturn(configuration).anyTimes();
+
+    List<TopicMetadata> topicMetadatas = new ArrayList<TopicMetadata>();
+    TopicMetadataResponse topicMetadataResponse = EasyMock.createMock(TopicMetadataResponse.class);
+    mocks.add(topicMetadataResponse);
+    EasyMock.expect(topicMetadataResponse.topicsMetadata()).andReturn(topicMetadatas);
+    
+    SimpleConsumer simpleConsumer = EasyMock.createMock(SimpleConsumer.class);
+    mocks.add(simpleConsumer);
+    EasyMock.expect(simpleConsumer.clientId()).andReturn(DUMMY_VALUE).times(2);
+    EasyMock.expect(simpleConsumer.send((TopicMetadataRequest)EasyMock.anyObject())).andThrow(new RuntimeException("No TopicMD"));
+    EasyMock.expect(simpleConsumer.send((TopicMetadataRequest)EasyMock.anyObject())).andReturn(topicMetadataResponse);
+    simpleConsumer.close();
+    EasyMock.expectLastCall().andVoid().anyTimes();
+
+    EasyMock.replay(mocks.toArray());
+    
+    EtlInputFormat inputFormat = new TestEtlInputFormat(simpleConsumer);    
+    List<TopicMetadata> actualTopicMetadatas = inputFormat.getKafkaMetadata(jobContext);
+    
+    EasyMock.verify(mocks.toArray());
+    
+    assertEquals(actualTopicMetadatas, topicMetadatas);
+  }
+
+  @Test(expected = RuntimeException.class)
```

Comment:
If you throw new RuntimeException("Failed to obtain metadata!", savedException),
then the exception message should just be "Failed to obtain metadata!", right?
So can you just catch it, and do assertEqual(e.getMessage(), "Failed to obtain metadata!") ?